docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,89 @@
+# Charm Architecture
+
+At its core, [Discourse](https://www.discourse.org/) is a [Ruby on Rails](https://rubyonrails.org/) application that integrates with [PostgreSQL](https://www.postgresql.org/) and [Redis](https://redis.io/).
```

Comment:
I think we should link to the postgresql-k8s charm and the redis-k8s charm rather than upstream projects, as it's those specific things that we integrate with/relate to.

Reply:
I intended this paragraph to be more specifically about Discourse itself, not the charm. Maybe that doesn't make a lot of sense?

Reply:
Ok, sounds good to me